### PR TITLE
Add animation drawable (animated vector drawable) support

### PR DESCRIPTION
**Pull request by S-H-Y-A**
### First time contributor checklist

- [x] I have read [how to contribute](/.github/CONTRIBUTING.md) to this project
- [x] I have read [the code of conduct](/.github/CODE_OF_CONDUCT.md) to this project

### Contributor checklist

- [x] I am using the provided [codeStyleConfig.xml](/.idea/codeStyles)
- [x] I have tested my contribution on these devices:
 * Asus ZenFone Max Pro (M2) Android 10 (Bata)
- [x] My contribution is fully baked and ready to be merged as is

----------

### Description

If Animation Drawable is set, when changing the Fab toggle, it will be animated.

Fab's rotate angle must be set to 0 if animation drawable is set.
The duration of animated drawable must be 200.

https://user-images.githubusercontent.com/74596628/111263157-bf5a0280-8668-11eb-9d45-b189c12353b3.mp4

### CODE REVIEW

**Comment by leinardi** [comment]
Hi @S-H-Y-A, thank you for your contribution. Could you please target `master` instead of the release branch? I'm updating the release branch only when a new release is out.

**Comment by S-H-Y-A** [comment]
Sorry, I fixed it.

File: library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java [hunk 1]
```diff
@@ -816,7 +835,23 @@ private void updateMainFabDrawable(boolean animate) {
         } else {
             UiUtils.rotateBackward(mMainFab, animate);
             if (mMainFabClosedDrawable != null) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && mMainFabClosedDrawable instanceof AnimatedVectorDrawable) {
+                    updated = true;
+                    mMainFab.setImageDrawable(mMainFabClosedDrawable);
+                    ((AnimatedVectorDrawable) mMainFabClosedDrawable).start();
+                } else if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N && mMainFabClosedDrawable instanceof AnimatedVectorDrawableCompat) {
```
Comment (leinardi) [inline library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java:834]: @S-H-Y-A I can't check this myself right now: are you 100% positive that on `VERSION_CODES.N` it is not possible to have `AnimatedVectorDrawableCompat`? What I'm asking is if we could have a case where the `VERSION.SDK_INT` is  `Build.VERSION_CODES.N` but `mMainFabClosedDrawable` is an instance of `AnimatedVectorDrawableCompat`.

(probably not, but I just want to be sure)

File: library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java [hunk 2]
```diff
@@ -805,8 +808,24 @@ private void toggle(boolean show, boolean animate) {
     }
 
     private void updateMainFabDrawable(boolean animate) {
+        boolean updated = false;
         if (isOpen()) {
             if (mMainFabOpenedDrawable != null) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && mMainFabOpenedDrawable instanceof AnimatedVectorDrawable) {
+                    updated = true;
```
Comment (leinardi) [inline library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java]: Could you please move

```
                    updated = true;
                    mMainFab.setImageDrawable(mMainFabOpenedDrawable);
```

outside the if/else here and on the else branch? If you move this code above the it, we could avoid some code duplication.

Something like this (I did not test it):
```java
                updated = true;
                mMainFab.setImageDrawable(mMainFabOpenedDrawable);
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && mMainFabOpenedDrawable instanceof AnimatedVectorDrawable) {
                    ((AnimatedVectorDrawable) mMainFabOpenedDrawable).start();
                } else if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N && mMainFabOpenedDrawable instanceof AnimatedVectorDrawableCompat) {
                    ((AnimatedVectorDrawableCompat) mMainFabOpenedDrawable).start();
                } else if (mMainFabOpenedDrawable instanceof AnimationDrawable){
                    ((AnimationDrawable) mMainFabOpenedDrawable).start();
                }
```

**Review by leinardi — CHANGES_REQUESTED** [review]

File: library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java (hunk 2 above)
Comment (S-H-Y-A) [reply library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java]: removed updated.

File: library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java (hunk 1 above)
Comment (S-H-Y-A) [reply library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java:834]: Yes, it works.

Look at this.
AnimatedVectorDrawableCompat

Comment (leinardi) [reply library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java:834]: Thanks :+1:

File: library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java (hunk 2 above)
Comment (leinardi) [reply library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java]: Just curious: why did you remove `updated`? I just asked to move it outside the if, not to remove it completely. Was it not needed?

Anyway, there is still code duplication for `mMainFab.setImageDrawable(mMainFabOpenedDrawable);`: please move it outside the if/else.

Comment (S-H-Y-A) [reply library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java]: Yes, `updated` was not needed.
I changed the code. Check please.

Comment (leinardi) [reply library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java]: Thanks!

**Review by leinardi — APPROVED** [review]

**Comment by S-H-Y-A** [comment]
This feature solved #132.

**Pull request merged**